### PR TITLE
Update Deps; Mocha + Chai

### PR DESCRIPTION
**Pull request by bkendall**
updating some dependencies, fixing some bugs, replaced `lab` and `code` with `mocha` and `chai`, improve coverage

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 node_modules/
+coverage
```
Comment (cflynn07) [inline .gitignore]: `coverage*` ?

File: .travis.yml [hunk 2]
```diff
@@ -3,12 +3,13 @@ node_js:
   - "0.10"
   - "0.12"
   - "4"
```
Comment (tjmehta) [inline .travis.yml]: 4.2 or 4.1 may be better

**Comment by tjmehta** [comment]
LGTM

File: .travis.yml (hunk 2 above)
Comment (bkendall) [inline .travis.yml]: I'm operating under the assumption that if 4.x will behave functionally the same for this project. If something changes drastically enough that this breaks, it'll be on a major version... then again, 4.2.x is the LTS release... maybe I just go with that in the future?

**Pull request merged**